flight/Modules/Stabilization/stabilization.c

```diff
@@ -390,16 +591,29 @@ static void stabilizationTask(void* parameters)
 	FlightStatusData flightStatus;
 	SystemSettingsAirframeTypeOptions airframe_type;
 
+	volatile bool settings_flag = true;
+	volatile bool flightstatus_flag = true;
+
 	float *actuatorDesiredAxis = &actuatorDesired.Roll;
 	float *rateDesiredAxis = &rateDesired.Roll;
 	smoothcontrol_initialize(&rc_smoothing);
 
 	// Connect callbacks
-	FlightStatusConnectCallbackCtx(UAVObjCbSetFlag, &flightStatusUpdated);
-	SystemSettingsConnectCallbackCtx(UAVObjCbSetFlag, &systemSettingsUpdated);
+	FlightStatusConnectCallbackCtx(UAVObjCbSetFlag, &flightstatus_flag);
+	SystemSettingsConnectCallbackCtx(UAVObjCbSetFlag, &settings_flag);
 	StabilizationSettingsConnectCallbackCtx(UAVObjCbSetFlag, &settings_flag);
 	VbarSettingsConnectCallbackCtx(UAVObjCbSetFlag, &settings_flag);
 	SubTrimSettingsConnectCallbackCtx(UAVObjCbSetFlag, &settings_flag);
+#ifdef TARGET_MAY_HAVE_BARO
+	AltitudeHoldSettingsConnectCallbackCtx(UAVObjCbSetFlag, &settings_flag);
+#else
+	/*
+	 * Sanity check uses this to know whether althold mode is permitted,
+	 * so we'd better not have a baro if the target doesn't "support"
+	 * one.
+	 */
+	PIOS_Assert(!PIOS_SENSORS_IsRegistered(PIOS_SENSOR_BARO));
```

Comment:
Would be better to put this into the sanity check IMO.

Reply:
The sanity check is in a library and is built for all targets.  It doesn't know if MAY_HAVE_BARO is true, so there's no choice but to do it here.

Reply:
Makes sense. :+1: